src/alto-ui/Datagrid/Datagrid.js

```diff
@@ -91,56 +91,48 @@ class Datagrid extends React.PureComponent {
     super(props);
 
     this.state = {
-      loaded: false,
+      loaded: true,
       collapsedGroups: {},
       resizer: initialStateResizer,
       columnsWidth: {},
+
     };
 
-    this.scrollingElt = null;
-    this.timeout = null;
-    this.initialTimeout = null;
+    this.isMacOS = isMacOSHelper();
 
     this.setStaticHeaderNode = this.setStaticHeaderNode.bind(this);
     this.setFrozenRowsNode = this.setFrozenRowsNode.bind(this);
     this.setStaticRowsNode = this.setStaticRowsNode.bind(this);
-    this.handleScrollXStaticHeader = this.handleScrollXStaticHeader.bind(this);
-    this.handleScrollYFrozenRows = this.handleScrollYFrozenRows.bind(this);
-    this.handleScrollStaticRows = this.handleScrollStaticRows.bind(this);
     this.handleMouseEnterResizeHandle = this.handleMouseEnterResizeHandle.bind(this);
     this.handleToggleGroup = this.handleToggleGroup.bind(this);
     this.handleStopResize = this.handleStopResize.bind(this);
     this.handleStartResize = this.handleStartResize.bind(this);
-    this.trackDimensions = debounce(this.trackDimensions.bind(this), WINDOW_RESIZE_DEBOUNCE);
 
     this.containerRef = React.createRef();
+
+    this.setScrollNode = this.setScrollNode.bind(this);
+    this.setScrollHeaderNode = this.setScrollHeaderNode.bind(this);
+    this.scrollListener = this.isMacOS
```

Comment:
do we need to put the condition here? can't we have on on everything on this.scrollListener.bind(this) ?

Reply:
IMO it's a better place.
Constructor runs only once and this condition will be checked once.
In your proposition condition will be checked on scroll